Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp

```diff
@@ -1944,6 +1944,8 @@ void WebLocalFrameLoaderClient::getLoadDecisionForIcons(const Vector<std::pair<W
 
 void WebLocalFrameLoaderClient::broadcastMainFrameURLChangeToOtherProcesses(const URL& url)
 {
+    if (auto* coreFrame = m_frame->coreFrame(); coreFrame && !coreFrame->settings().siteIsolationEnabled())
```

Comment:
`RefPtr`
Also it seems we could create a helper function `WebLocalFrameLoaderClient::isSiteIsolationEnabled()` to avoid duplicate code.